apps/ejabberd/src/mod_roster.erl

```diff
@@ -397,10 +410,23 @@ do_process_item_set(JID1,
                     To,
                     #xmlel{attrs = Attrs, children = Els}) ->
     LJID = jid:to_lower(JID1),
+    Item = get_roster_by_jid(LUser, LServer, LJID),
```

Comment:
My understanding was that the transaction was here mostly to protect the sequence of actions performed to update a roster (update roster, run hook, write roster version), so I did not bother much about roster get. I take your point, though, and await your recommendation - put "get" back into a transaction, or get rid of transaction altogether?